src/glfw-sandbox/main.cpp

```diff
@@ -9,136 +9,31 @@
-
-    // Lighting
-    glEnable(GL_NORMALIZE);
-    glEnable(GL_AUTO_NORMAL);
-    glShadeModel(GL_SMOOTH);
-    glLightModeli(GL_LIGHT_MODEL_TWO_SIDE, 1);
-
-    // Texture related
-    glPixelStorei(GL_PACK_ALIGNMENT, 1);
-
-    // Blending
-    glBlendFunc(GL_ONE, GL_ONE);
-
-    glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_NICEST);
-
-    glViewport(0, 0, width, height);            // Set viewport region.
-    glMatrixMode(GL_PROJECTION);                // Switch to Projection-Matrix mode.
-    glLoadIdentity();                           // Reset the Projection-Matrix.
-
-    gluPerspective(fov, width / (double)height, 0.09, 25);
-
-    glMatrixMode(GL_MODELVIEW);                 // Select the Modelview-Matrix
-}
-
-void drawGrid(float xSize, float zSize)
-{
-    glColor3ub(255, 255, 255);
-    glLineWidth(1);
-
-    glBegin(GL_LINES);
-    for (int i = -zSize / 2.0; i <= zSize / 2.0; i++) {
-        glVertex3f(-xSize/2.0, 0, i);
-        glVertex3f(xSize/2.0, 0, i);
-    }
-
-    for (int i = -xSize / 2.0; i <= xSize / 2.0; i++) {
-        glVertex3f(i, 0, -zSize / 2.0);
-        glVertex3f(i, 0, zSize / 2.0);
-    }
-
-    glEnd();
-
-    glLineWidth(2);
-
-    glBegin(GL_LINES);
-        glColor3ub(255, 0, 0);
-        glVertex3f(0, 0, 0);
-        glVertex3f(1, 0, 0);
-
-        glColor3ub(0, 255, 0);
-        glVertex3f(0, 0, 0);
-        glVertex3f(0, 1, 0);
-
-        glColor3ub(0, 0, 255);
-        glVertex3f(0, 0, 0);
-        glVertex3f(0, 0, 1);
-    glEnd();
-
-    glColor3ub(255, 255, 255);
-
-}
-
-void drawTriangle()
-{
-    glBegin(GL_TRIANGLES);
-        glColor3ub(255, 0, 0);
-        glVertex3f(0, 0, 0);
-
-        glColor3ub(0, 255, 0);
-        glVertex3f(1, 0, 0);
-
-        glColor3ub(0, 0, 255);
-        glVertex3f(0.5, 1, 0);
-    glEnd();
-}
-
-void drawWheel(int lod) {
-    glBegin(GL_POLYGON);
-
-        glVertex3f(0, 0, 0);
-
-        for (int i = 0; i <= lod; i++)
-        {
-            glVertex3f(std::cos((i * 360 / (float)lod) * 3.14/180.0), std::sin((i * 360 / (float)lod)* 3.14/180.0), 0);
-        }
-
-    glEnd();
-}
-
 int main() {
 
-    GLFWwindow* window;
-
-    if (!glfwInit())
-    {
-        std::cout << "Failed to initialize GLFW library" << std::endl;
-        return 1;
-    }
+    drawObjects drawx;
+    windowCreator windowObject(1280, 720, 60);
+    auto window = windowObject.getWindow();
+    
 
-    const int resolutionWidth = 1280;
-    const int resolutionHeight = 720;
-    window = glfwCreateWindow(resolutionWidth, resolutionHeight,    // Resolution
-                              "GLFW playground",                    // Title
-                              NULL,                                 // Monitor for fullscreen
-                              NULL);                                // Window to steal OpenGL context from - in most cases NULL
+    // What is the difference between this and the previous "GLFWwindow* window;" call ???
+    /*
+    
+    struct window {

```

Comment:
Struct is something like a class, but from the ancient times :) It helps you to pack data together.

Reply:
`GLFWwindow *window` is just a declaration.

`struct window {...};` means "I would like to keep everything related to window on one place, and have width, height, title, monitor and ... glfwwindow pointer packed in one struct (=something like a class, will explain on call).